File: Noise.Infrastructure/RemoteDto/RoSearchResultItem.cs
using System;
using System.Runtime.Serialization;
using Noise.Infrastructure.Dto;

namespace Noise.Infrastructure.RemoteDto {
	[DataContract]
	public class RoSearchResultItem {
		[DataMember]
		public long		TrackId { get; set; }
		[DataMember]
		public string	TrackName { get; set; }
		[DataMember]
		public long		AlbumId { get; set; }
		[DataMember]
		public string	AlbumName { get; set; }
		[DataMember]
		public long		ArtistId { get; set; }
		[DataMember]
		public string	ArtistName { get; set; }
		[DataMember]
		public bool		CanPlay { get; set; }

		public RoSearchResultItem( SearchResultItem src ) {
            ArtistId = src.Artist?.DbId ?? Constants.cDatabaseNullOid;
            ArtistName = src.Artist != null ? src.Artist.Name : String.Empty;
            AlbumId = src.Album?.DbId ?? Constants.cDatabaseNullOid;
            AlbumName = src.Album != null ? src.Album.Name : String.Empty;
            TrackId = src.Track?.DbId ?? Constants.cDatabaseNullOid;
            TrackName = src.Track != null ? src.Track.Name : String.Empty;

            CanPlay = false;
        }
	}
}

File: Noisie/Noisie/Infrastructure/Dto/RoSearchResultItem.m
//
//  RoSearchResultItem.m
//  Noisie
//
//  Created by William Swanson on 11/18/11.
//  Copyright (c) 2011 Secret Squirrel Products. All rights reserved.
//

#import "RoSearchResultItem.h"

@implementation RoSearchResultItem

@synthesize TrackId;
@synthesize TrackName;
@synthesize AlbumId;
@synthesize AlbumName;
@synthesize ArtistId;
@synthesize ArtistName;
@synthesize CanPlay;

- (void) dealloc {
    self.TrackId = nil;
    self.TrackName = nil;
    self.AlbumId = nil;
    self.AlbumName = nil;
    self.ArtistId = nil;
    self.ArtistName = nil;
    self.CanPlay = nil;
    
    [super dealloc];
}

- (NSString *) formattedTitle {
    NSString    *retValue = @"";
    
    if([self.TrackName length] > 0 ) {
        retValue = [NSString stringWithFormat:@"Track: %@ (%@/%@)", self.TrackName, self.ArtistName, self.AlbumName];
    }
    else if([self.AlbumName length] > 0 ) {
        retValue = [NSString stringWithFormat:@"Album: %@ (%@)", self.AlbumName, self.ArtistName];
    }
    else if([self.ArtistName length] > 0 ) {
        retValue = [NSString stringWithFormat:@"Artist: %@", self.ArtistName];
    }
    else {
        retValue = [NSString stringWithFormat:@"%@", @"(Nothing was found.)"];
    }
    
    return( retValue );
}
@end
